### ISSUE TITLE
Refactoring e2e tests focus/skip logic

### ISSUE THREAD

**Issue by fossedihelm** [kubevirt/kubevirt#9041]
**Is your feature request related to a problem? Please describe**:
Currently, e2e tests run in different sigs (`compute`/`storage`/ etc..).
How we discriminate whether or not a test is part of a particular `sig` is determined by a particular "tag" in the description of each test (`[sig-compute]`/`[sig-storage]`/ etc..).
For this, we use [description based filtering](https://onsi.github.io/ginkgo/#description-based-filtering).
With the growth of the project, this logic becomes hard to maintain and scale.
Also, some tags can be removed as shown here https://github.com/kubevirt/kubevirt/pull/8799

**Describe the solution you'd like**:
Ginkgo provides [label decorator](https://onsi.github.io/ginkgo/#the-label-decorator), which allows us to leave the description free from predefined tags, making the code cleaner and gives the possibility to write more complex logic with logical expressions.
Since the change affects the whole e2e suite, it is better to do things step by step:

- [x] Swap using "label decorator" instead of "description based filtering" https://github.com/kubevirt/kubevirt/pull/8881
- [ ] Analyze and design what labels we need
- [ ] Move to the new labels and logic

**Comment by fossedihelm** [comment]
Any suggestions or ideas on new labels (and how to subdivide the various tests to different lines) are greatly appreciated.
Thanks

**Comment by xpivarc** [comment]
Thanks, Federico! I would just add that we should strive for one additional goal and that is programmatical skips. These are often error-prone. The workflow we should provide is "I want to test these features, maybe I don't have HW for this so I am OK to exclude these tests and I expect to be not surprised".

**Comment by xpivarc** [comment]
I think it will come down to labels in at least 3 categories: sigs, features, hw requirements.

**Comment by dhiller** [comment]
I agree with @xpivarc that there are (implicit) categories, which IMHO should be made explicit, i.e. by adding a prefix or suffix to show the category, since to my knowledge Ginkgo doesn't have label categorization.

As I see it `sig` labels currently have two functions:
* make test ownership clear
* act as a broader selection mechanism for tests

In contrast to this we have requirement labels like `sriov` and `vgpu` or features like 'root'

What I am missing currently is a clear naming and explicit usage of labels to what is being wanted to achieve, and I'd really like to see this happen in the future, so everyone understands, i.e. why a test has a `sig-compute` label but actually is not found in the sig-compute lane run.

**Comment by dhiller** [comment]
I.e maybe something like using explicit env vars on job definitions like:

```yaml
  env:
  - name: KUBEVIRT_E2E_REQUIRES
    value: "vgpu-req,storage-req"
  - name: KUBEVIRT_E2E_OWNERSHIP
    value: "sig-storage"
```

(I'm obviously making things up here, maybe not spot on)

**Comment by dhiller** [comment]
So in the above example the ownership would make clear who owns the lane (which currently is a problem as I see it), what the requirements are, and the third one would select the tests that are run on this lane. I guess that the latter might be implicit and redundant.

**Comment by dhiller** [comment]
@fossedihelm I think this one is not yet done, right?

/remove-lifecycle rotten
/lifecycle frozen

**Comment by fossedihelm** [comment]
@dhiller Yes, correct! Thanks
